katka/serializer_fields.py

```diff
@@ -0,0 +1,52 @@
+from katka.constants import STATUS_ACTIVE
+from katka.models import Credential, Team
+from rest_framework import serializers
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.relations import PrimaryKeyRelatedField
+
+
+class GroupNameField(serializers.RelatedField):
+    def to_internal_value(self, group_name):
+        qs = self.get_queryset()
+        try:
+            return qs.get(name=group_name)
+        except qs.model.DoesNotExist:
+            raise NotFound
+
+    def to_representation(self, group):
+        return group.name
+
+
+class TeamRelatedField(PrimaryKeyRelatedField):
+    def get_queryset(self):
+        """Only get the teams that are connected to a group that the user is a member of"""
+        return Team.objects.filter(group__in=self.context['request'].user.groups.all(), status=STATUS_ACTIVE)
+
+    def fail(self, key, **kwargs):
+        """
+        Instead of replying with a 400 with a message 'does_not_exist', we would rather tell the user that they
+        do not have permission because the referred team either does not exist or they are not part of that team.
+        """
+        if key == 'does_not_exist':
+            raise PermissionDenied('User is not a member of this group')
```

Comment:
Maybe it is nice to add 'group might not exist' to the error, might be useful when debugging

Reply:
I updated the messages and put the method in a mixin, because more views will probably use this next